src/main/scala/com/horizon/exchangeapi/auth/Module.scala

```diff
@@ -0,0 +1,91 @@
+package com.horizon.exchangeapi.auth
+
+import com.horizon.exchangeapi._
+import javax.security.auth._
+import javax.security.auth.callback._
+import javax.security.auth.spi.LoginModule
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.util.Try
+
+class Module extends LoginModule with AuthSupport {
+  private var subject: Subject = _
+  private var handler: CallbackHandler = _
+  private var identity: Identity = _
+  lazy val logger: Logger = LoggerFactory.getLogger(ExchConfig.LOGGER)
+
+
+  override def initialize(
+    subject: Subject,
+    handler: CallbackHandler,
+    sharedState: java.util.Map[String, _],
+    options: java.util.Map[String, _]
+  ): Unit = {
+    this.subject = subject
+    this.handler = handler
+  }
+
+  override def login(): Boolean = {
```

Comment:
This is where the actual login logic is performed, and is called by the `LoginContext` when its `login` method is called. This uses the callback to get acces to the web request, and then uses the logic from the `credsAndLog` to get an Identity from the request. This is later attached to the subject in the `commit` method, which is called by the context after login succeeds, and that is where we can get access to it in the route handling code.